gobblin-core/src/main/java/gobblin/converter/EnvelopeSchemaConverter.java

```diff
@@ -0,0 +1,111 @@
+package gobblin.converter;
+
+import com.google.common.base.Optional;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.filter.AvroProjectionConverter;
+import gobblin.converter.filter.AvroSchemaFieldRemover;
+import gobblin.metrics.kafka.KafkaAvroSchemaRegistry;
+import gobblin.metrics.kafka.SchemaRegistryException;
+import gobblin.util.AvroUtils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.ExecutionException;
+import javax.xml.bind.DatatypeConverter;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+
+/**
+ * A converter for opening GoldenGate envelope schema.
+ */
+public class EnvelopeSchemaConverter extends Converter<Schema, String, GenericRecord, GenericRecord> {
+
+  public static final String PAYLOAD_SCHEMA_ID = "envelope.schemaIdField";
```

Comment:
Can you change these two keys' names, e.g. "EnvelopeSchemaConverter.payloadSchemaIdField"? It is more clear to see where this key belongs to.